intera_interface/src/intera_motion_interface/motion_controller_action_client.py

```diff
@@ -54,12 +54,17 @@ def send_trajectory(self, trajectory):
         goal.trajectory = trajectory
         self._client.send_goal(goal)
 
-    def wait_for_result(self):
+    def wait_for_result(self, timeout=None):
         """
         Wait for the current task to finish and then return the result
+        @param timeout: maximum time to block before returning
         @return: the result of the MotionCommand.action
         """
-        self._client.wait_for_result()
+        if timeout is None:
+            rosTimeout = rospy.Duration()
+        else:
+            rosTimeout = rospy.Duration(timeout)
+        self._client.wait_for_result(rosTimeout)
         return self._client.get_result()
```

Comment:
What result is returned when the timeout is triggered? For example, if the previous trajectory was successful, will it return the previous message? @asthakukreja  - can you confirm that this will send back either `None` when the timeout is triggered? If not, then can you explicitly have the code return `None` when the timeout is triggered?

Reply:
The result is 'True' if the goal finished successfully. And it returns 'None' if the goal didn't finish within the allocated timeout.